Core/src/org/sleuthkit/autopsy/geolocation/datamodel/GeolocationUtils.java

```diff
@@ -0,0 +1,86 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.geolocation.datamodel;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+
+/**
+ * GeolocationUtilis class for common to be share across in the package
+ *
+ */
+final class GeolocationUtils {
+
+    /**
+     * This is a list of attributes that are related to a geolocation waypoint
+     * but are for information\artifact properties purpose. They are not needed
+     * for the placement of a point on a map;
+     */
+    private static final BlackboardAttribute.ATTRIBUTE_TYPE[] OTHER_GEO_ATTRIBUTES = {
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_LOCATION,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME_START,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME_END,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME_CREATED,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_VELOCITY,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_BEARING,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_HPRECISION,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_VPRECISION,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_MAPDATUM,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PROG_NAME,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PATH_SOURCE,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_FLAG,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DEVICE_MAKE,
+        BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DEVICE_MODEL
+    };
+
+    /**
+     * This is a Utility class that should not be constructed.
+     */
+    private GeolocationUtils() {
+
+    }
+
+    /**
+     * Get a list of Waypoint.Property objects for the given artifact.
+     *
+     * @param artifact Blackboard artifact to get attributes\properties from
+     *
+     * @return A List of Waypoint.Property objects
+     *
+     * @throws GeoLocationDataException
+     */
+    static List<Waypoint.Property> getOtherGeolocationProperties(BlackboardArtifact artifact) throws GeoLocationDataException {
+        List<Waypoint.Property> list = new ArrayList<>();
+
+        for (BlackboardAttribute.ATTRIBUTE_TYPE type : OTHER_GEO_ATTRIBUTES) {
+            String key = type.getDisplayName();
+            String value = ArtifactUtils.getString(artifact, type);
+
+            if (value == null) {
+                value = "";
+            }
```

Comment:
I still don't understand why we want to display empty properties. Why, for example, should every geolocation artifact be expected to display properties corresponding to TSK_DEVICE_MAKE and TSK_DEVICE_MODEL when these are specific to EXIF artifacts? 

I can't help but feel inferior code in KMLReport is in the driver's seat here. As noted above, I think we need to take a step back and take a hard look at what we need the common code / data model for the two types of geolocation data displays to do. I think the answer is we need something that will go to the case database, fetch all the artifacts that can be uniformly encapsulated as waypoints, adn return them as waypoints. Every waypoint shoul dhave some fixed properties and some variable properties. If the artifact is missing the fixed properties, it is not a valid waypoint and should be discarded. The wayppoints should not cook up empty properties for the sake of the views - the views need to do what they will with the optional properties that exist.